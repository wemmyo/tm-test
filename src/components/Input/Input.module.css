.inputGroup {
  margin-bottom: 1rem;
}
.label {
  display: inline-block;
  margin-bottom: 0.5rem;
}
.input {
  /* display: block; */
  width: 100%;
  padding: 0.6em 1em;
  background: #fff;
  border: 1px solid rgba(34, 36, 38, 0.15);
  color: rgba(0, 0, 0, 0.87);
  border-radius: 0.28rem;
  transition: box-shadow 0.1s ease, border-color 0.1s ease;
  box-sizing: border-box;
}
