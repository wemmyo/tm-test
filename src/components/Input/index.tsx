import * as React from "react";
import styles from "./Input.module.css";

interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: string;
}

const Input: React.FC<InputProps> = ({ label, ...props }) => {
  return (
    <div className={styles.inputGroup}>
      {label && <label className={styles.label}>{label}</label>}
      <input {...props} className={styles.input} />
    </div>
  );
};

export default Input;
