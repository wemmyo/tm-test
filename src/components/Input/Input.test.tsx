import React from "react";
import { shallow } from "enzyme";
import Input from "./index";

it("renders without crashing", () => {
  expect(shallow(<Input label="Test label" />)).toMatchSnapshot();
});
