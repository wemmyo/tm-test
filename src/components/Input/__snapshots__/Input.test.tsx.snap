// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="inputGroup"
>
  <label
    className="label"
  >
    Test label
  </label>
  <input
    className="input"
  />
</div>
`;
